src/utils.ts

```diff
@@ -1,69 +1,75 @@
-    contracts: {
-      catalyst: {
-        address: '0x4a2f10076101650f40342885b99b6b101d83c486',
-        class: Catalyst
-      },
-      POIs: {
-        address: '0x0ef15a1c7a49429a36cb46d4da8c53119242b54e',
-        class: List
-      },
-      denylistedNames: {
-        address: '0x0c4c90a4f29872a2e9ef4c4be3d419792bca9a36',
-        class: List
-      }
-    }
+// eslint-disable-next-line
+export async function nameDenylistForProvider(ethereumProvider: any): Promise<ListContract | undefined> {
+  const rm = new RequestManager(ethereumProvider)
+  const networkId = (await rm.net_version()).toString()
+
+  if (!(networkId in denylistNamesDeployments))
+    throw new Error(
+      `There is no deployed NameDenylist contract for networkId=${networkId}`
+    )
+
+  try {
+    const contractAddress = denylistNamesDeployments[networkId]
+
+    return (await new ContractFactory(rm, listAbiItems).at(
+      contractAddress
+    )) as any as ListContract
+  } catch (error) {
+    return undefined
   }
 }
 
-export function handlerForNetwork(networkKey: string, contractKey: string) {
+// eslint-disable-next-line
+export async function poiListForProvider(ethereumProvider: any): Promise<ListContract | undefined> {
+  const rm = new RequestManager(ethereumProvider)
+  const networkId = (await rm.net_version()).toString()
+
+  if (!(networkId in poiDeployments))
+    throw new Error(
+      `There is no deployed PoiDenylist contract for networkId=${networkId}`
+    )
+
   try {
-    const provider = httpProviderForNetwork(networkKey)
-    const eth = new Eth(provider)
-    const network = networks[networkKey]
-    const contract = network.contracts[contractKey]
-    const address = Address.fromString(contract.address)
-    const contractInstance = new contract.class(eth, address)
+    const contractAddress = poiDeployments[networkId]
 
-    return {
-      provider,
-      network,
-      contract: contractInstance
-    }
+    return (await new ContractFactory(rm, listAbiItems).at(
+      contractAddress
+    )) as any as ListContract
   } catch (error) {
     return undefined
   }
 }
 
-export function httpProviderForNetwork(networkKey: string) {
-  const network = networks[networkKey]
-  const url = network.http
-  return new HttpProvider(url)
+// eslint-disable-next-line
+export async function catalystRegistryForProvider(ethereumProvider: any): Promise<CatalystContract | undefined> {
+  const rm = new RequestManager(ethereumProvider)
+  const networkId = (await rm.net_version()).toString()
+
+  if (!(networkId in catalystDeployments))
+    throw new Error(
+      `There is no deployed CatalystProxy contract for networkId=${networkId}`
+    )
+
+  try {
+    const contractAddress = catalystDeployments[networkId]
+
+    return (await new ContractFactory(rm, catalystAbiItems).at(
+      contractAddress
+    )) as any as CatalystContract
+  } catch (error) {
+    return undefined
```

Comment:
No need for logging here? Is it the caller's responsibility?

Reply:
oops

Reply:
I decided that since we are breaking compatibility, bubbling up the error (removing catch) is the best.